Mathlib/Data/Nat/Count.lean

```diff
@@ -146,6 +146,22 @@ theorem count_lt_card {n : ℕ} (hp : (setOf p).Finite) (hpn : p n) : count p n
   (count_lt_count_succ_iff.2 hpn).trans_le (count_le_card hp _)
 #align nat.count_lt_card Nat.count_lt_card
 
+theorem count_true (hp : ∀ n, p n) : count p = id := by
```

Comment:
Updated (the applied versions have slightly weaker hypotheses).

I'm tempted to keep the simp versions un-applied since `True` usually only shows up mid-`simp` and this way it will simp expressions like `Nat.nth (fun _ ↦ True) '' S`

Reply:
I see. It might or might not make sense to use `Set.image_eta` (unsure what it's called) explicitly in your simp call instead. It's the lemma stating `f '' s = (fun x => f x) '' s`, and it's really useful for exactly this kind of situation where your function is underapplied.

Reply:
The slightly unwieldy thing about that is that it requires looking up a lemma for each potential situation, e.g. for `Finset` instead of `Set`, etc; `eta_expand` works in general but is a bit awkward to do in the middle of a simp.

Occasionally I see `_ = id` simp-lemmas that have a corresponding `_ x = x` (sometimes also simp) lemma called `..._apply`; is that an OK pattern?

Reply:
Depends what the lemmas are, but can you tell that your approach requires many more lemmas than mine?

Reply:
I suspect the lemmas you are talking about are about bundled morphisms

Reply:
> I suspect the lemmas you are talking about are about bundled morphisms

Many of the hits on Loogle are indeed; sometimes they're plain functions e.g. [docs#Set.image_id_eq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Set.image_id_eq), [docs#Set.preimage_id_eq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Set.preimage_id_eq).

Either way, I'm happy to just make these the applied versions. It's straightforward enough to define the eta-reduced versions wherever needed.